.github/workflows/build-executable-test-deploy.yml

```diff
@@ -155,3 +155,232 @@ jobs:
+    runs-on: ubuntu-latest
+    needs:
+      - build-executable
+
+    steps:
+    - name: Checkout Code
+      uses: actions/checkout@v3
+      with:
+        submodules: true
+
+    - name: Download executables from the previous job
+      uses: actions/download-artifact@v2
+      with:
+        name: gprofiler
+        path: output/
+
+    # the tests need the gprofiler image built (from Dockerfile). I run it separately here, because "docker build" prints the build logs
+    # more nicely. the tests will then be able to use the built image.
+    - name: Build gProfiler image
+      run: ./scripts/build_x86_64_container.sh --build-arg EXE_PATH=output/gprofiler
+
+    - name: Export gProfiler image
+      run: mkdir -p output && docker image save gprofiler > output/gprofiler.img
+
+    - name: Upload the image artifact
+      uses: actions/upload-artifact@v2
+      with:
+        name: gprofiler.img
+        path: output/
+        retention-days: 1
+
+  test:
+    if: ${{ !startsWith(github.ref, 'refs/tags/') }}
+    needs: build-container-x64
+
+    runs-on: ubuntu-20.04  # for 3.10.0 python tag
+
+    strategy:
+      fail-fast: false  # helps detecting flakiness / errors specific to one Python version
+      matrix:
+        python-version:
+          - "3.8"
+          - "3.9"
+          - "3.10.0"  # see https://github.com/Granulate/gprofiler/issues/502, we need 3.10.0, others fail PyPerf tests.
+
+    steps:
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Install Java
+      uses: actions/setup-java@v1
+      with:
+        java-version: '8.0.275'
+        java-package: jdk
+        architecture: x64
+
+    - name: Install Node.JS
+      uses: actions/setup-node@v2
+      with:
+        # same version as used in tests/containers/nodejs/Dockerfile
+        node-version: 10.x
+
+    - name: Checkout Code
+      uses: actions/checkout@v3
+      with:
+        submodules: true
+
+    - name: Download the executable from previous job
+      uses: actions/download-artifact@v2
+      with:
+        name: gprofiler
+        path: output/
+
+    - name: Download the image from previous job
+      uses: actions/download-artifact@v2
+      with:
+        name: gprofiler.img
+        path: output/
+
+    - name: Add +x to gprofiler
+      run: chmod +x ./output/gprofiler
+
+    - name: Extract resources from gProfiler executable
+      run: sudo ./output/gprofiler extract-resources --resources-dest=./gprofiler/resources
+
+    # used in the tests
+    - name: Import gProfiler image
+      run: docker image load < output/gprofiler.img
+
+    # TODO: Add docker layer caching when GitHub Actions cache is stabilized and works good with "satackey/action-docker-layer-caching@v0.0.11"
+    - name: Run gProfiler tests
+      run: NO_APT_INSTALL=1 ./tests/test.sh --ignore=tests/test_executable.py
+
+  build-docker-x64:
```

Comment:
Isn't this job & `build-container-x64` the same? This seems like some duplication.

Reply:
Some steps of job `build-docker-x64` were the same as steps in `build-container-x64`, and I moved other steps from that job to `deploy-container` job